backend/posts/filters.py

```diff
@@ -0,0 +1,47 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django_filters import widgets, filters, rest_framework as rest_filters
+
+
+class StrictBooleanWidget(widgets.BooleanWidget):
+    """
+    Widget with functionality similar to BooleanWidget provided with django_filters
+    but that allows to handle invalid values instead of just ignoring them.
+    """
```

Comment:
Normally, `BooleanFilter` shipped with `django_filters` uses `BooleanWidget` which converts all invalid values for a boolean filter, such as "dog", "maybe" or "fasle" into `None` thus disabling the filter.

I don't really see any advantage to keep this behavior, so I created this `StrictBooleanWidget` class that, when used with an appropriate field class (below) allows me to response with `400 Bad Request` if the user provides something else than True, False, 0 or 1 or None.

(`django_filters` has also a second implementation of `BooleanFilter` that uses `NullBooleanField` and has the same behavior that I dislike)

Reply:
Seems legit for me. Thx for explanation 👍